ibm/service/codeengine/resource_ibm_code_engine_function.go

```diff
@@ -0,0 +1,845 @@
+		if err = d.Set("created_at", function.CreatedAt); err != nil {
+			return diag.FromErr(fmt.Errorf("error setting created_at: %s", err))
+		}
+	}
+	if !core.IsNil(function.Endpoint) {
+		if err = d.Set("endpoint", function.Endpoint); err != nil {
+			return diag.FromErr(fmt.Errorf("error setting endpoint: %s", err))
+		}
+	}
+	if !core.IsNil(function.EndpointInternal) {
+		if err = d.Set("endpoint_internal", function.EndpointInternal); err != nil {
+			return diag.FromErr(fmt.Errorf("error setting endpoint_internal: %s", err))
+		}
+	}
+	if err = d.Set("entity_tag", function.EntityTag); err != nil {
+		return diag.FromErr(fmt.Errorf("error setting entity_tag: %s", err))
+	}
+	if !core.IsNil(function.Href) {
+		if err = d.Set("href", function.Href); err != nil {
+			return diag.FromErr(fmt.Errorf("error setting href: %s", err))
+		}
+	}
+	if !core.IsNil(function.ID) {
+		if err = d.Set("function_id", function.ID); err != nil {
+			return diag.FromErr(fmt.Errorf("error setting function_id: %s", err))
+		}
+	}
+	if !core.IsNil(function.Region) {
+		if err = d.Set("region", function.Region); err != nil {
+			return diag.FromErr(fmt.Errorf("error setting region: %s", err))
+		}
+	}
+	if !core.IsNil(function.ResourceType) {
+		if err = d.Set("resource_type", function.ResourceType); err != nil {
+			return diag.FromErr(fmt.Errorf("error setting resource_type: %s", err))
+		}
+	}
+	if !core.IsNil(function.Status) {
+		if err = d.Set("status", function.Status); err != nil {
+			return diag.FromErr(fmt.Errorf("error setting status: %s", err))
+		}
+	}
+	statusDetailsMap, err := resourceIbmCodeEngineFunctionFunctionStatusToMap(function.StatusDetails)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	if err = d.Set("status_details", []map[string]interface{}{statusDetailsMap}); err != nil {
+		return diag.FromErr(fmt.Errorf("error setting status_details: %s", err))
+	}
+	if err = d.Set("etag", response.Headers.Get("Etag")); err != nil {
+		tfErr := flex.TerraformErrorf(err, fmt.Sprintf("Error setting etag: %s", err), "ibm_code_engine_function", "read")
+		return tfErr.GetDiag()
+	}
+
+	return nil
+}
+
+func resourceIbmCodeEngineFunctionUpdate(context context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	codeEngineClient, err := meta.(conns.ClientSession).CodeEngineV2()
+	if err != nil {
+		tfErr := flex.TerraformErrorf(err, err.Error(), "ibm_code_engine_function", "update")
+		log.Printf("[DEBUG]\n%s", tfErr.GetDebugMessage())
+		return tfErr.GetDiag()
+	}
+
+	updateFunctionOptions := &codeenginev2.UpdateFunctionOptions{}
+
+	parts, err := flex.SepIdParts(d.Id(), "/")
+	if err != nil {
+		tfErr := flex.TerraformErrorf(err, err.Error(), "ibm_code_engine_function", "update")
+		return tfErr.GetDiag()
+	}
+
+	updateFunctionOptions.SetProjectID(parts[0])
+	updateFunctionOptions.SetName(parts[1])
+
+	hasChange := false
+
+	patchVals := &codeenginev2.FunctionPatch{}
+	if d.HasChange("project_id") {
```

Comment:
this block is n't required when you mark ForceNew : true on "project_id" argument